coriolis/tests/test_wsman.py

```diff
@@ -0,0 +1,157 @@
+            "127.0.0.1", 5986)
+        mock_connect.assert_called_once_with(
+            url="https://127.0.0.1:5986/wsman", username="user",
+            password="pass", cert_pem=None, cert_key_pem=None)
+        self.assertIsInstance(result, self.conn.__class__)
+
+    def test_from_connection_info_missing_keys(self):
+        self.assertRaises(ValueError, self.conn.from_connection_info,
+                          {"username": "user", "password": "pass"})
+
+    def test_from_connection_info_invalid_type(self):
+        self.assertRaises(ValueError, self.conn.from_connection_info,
+                          'invalid-connection-type')
+
+    def test_disconnect(self):
+        self.conn.disconnect()
+        self.assertIsNone(self.conn._protocol)
+
+    def test_set_timeout(self):
+        self.conn.set_timeout(self.conn._conn_timeout)
+        self.assertEqual(self.conn._protocol.transport.timeout,
+                         self.conn._conn_timeout)
+
+    def test__exec_command(self):
+        self.conn._protocol.get_command_output.return_value = (
+            "std_out", "std_err", 0)
+        std_out, std_err, exit_code = self.conn._exec_command(
+            self.cmd, self.args)
+        self.assertEqual(std_out, "std_out")
+        self.assertEqual(std_err, "std_err")
+        self.assertEqual(exit_code, 0)
+
+        self.conn._protocol.open_shell.assert_called_once_with(
+            codepage=wsman.CODEPAGE_UTF8)
+        shell_id = self.conn._protocol.open_shell.return_value
+        self.conn._protocol.run_command.assert_called_once_with(
+            shell_id, self.cmd, self.args)
+        command_id = self.conn._protocol.run_command.return_value
+        self.conn._protocol.get_command_output.assert_called_once_with(
+            shell_id, command_id)
+
+    def test__exec_command_exception(self):
+        self.conn._protocol.get_command_output.side_effect = requests.\
+            exceptions.ReadTimeout
+        self.assertRaises(exception.OSMorphingWinRMOperationTimeout,
+                          self.conn._exec_command, self.cmd, self.args)
+
+    def test_exec_command(self):
+        self.conn._protocol.get_command_output.return_value = (
+            "std_out", "std_err", 0)
+        std_out = self.conn.exec_command(self.cmd, self.args)
+        self.assertEqual(std_out, "std_out")
+
+    def test_exec_command_exception(self):
+        self.conn._protocol.get_command_output.return_value = (
+            "std_out", "std_err", 1)
+        self.assertRaises(exception.CoriolisException, self.conn.exec_command,
+                          self.cmd, self.args)
+
+    def test_exec_ps_command(self):
+        self.conn.exec_command = mock.Mock()
+        self.conn.exec_command.return_value = "std_out\n\n"
+        result = self.conn.exec_ps_command(self.cmd)
+        self.conn.exec_command.assert_called_once_with(
+            "powershell.exe", ["-EncodedCommand", mock.ANY], timeout=None)
```

Comment:
Please add the encoded command string instead of `mock.ANY`

`'dABlAHMAdABfAGMAbQBkAA=='`